### PR TITLE
initial commit for app store refactor

### PR DESCRIPTION
**Pull request by sorja**
part of #905 : app

### CODE REVIEW

File: webapp/app/assessment/assessmentState.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const _isLocked = (type: any) => R.pipe(getStateAssessment(type), R.propOr(true,
 // ======  Lock functions
 
 export const isLocked = (assessment: any) => (state: any) => {
```
Comment (minotogna) [inline webapp/app/assessment/assessmentState.ts:42]: better to refactor these to hooks

File: webapp/app/assessment/assessmentState.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export const isLocked = (assessment: any) => (state: any) => {
 }
 
 export const canToggleLock = (assessment: any) => (state: any) => {
```
Comment (minotogna) [inline webapp/app/assessment/assessmentState.ts:54]: better to refactor this to hook

Comment (sorja) [reply webapp/app/assessment/assessmentState.ts:54]: should they be done in this pr or in assessment refactor?

File: webapp/app/assessment/fra/fraState.ts [hunk 3]
```diff
@@ -17,24 +17,26 @@ export const isLocked = (state: any) => AssessmentState.isLocked(getAssessment(s
...
-  }
-  const userInfo = UserState.getUserInfo(state)
-  const countryIso = AppState.getCountryIso(state)
-
-  // if user is collaborator, he could have restricted access to specific tables
-  if (isCollaborator(countryIso, userInfo)) {
-    const assessment = getAssessment(state)
-    const allowedTables = Assessment.getTablesAccess(assessment)
-    return isCollaboratorAllowedToEditSectionData(section, allowedTables)
+const canEditSection =
```
Comment (minotogna) [inline webapp/app/assessment/fra/fraState.ts:20]: better to refactor to hook

File: webapp/app/country/actions.ts [hunk 4]
```diff
@@ -2,11 +2,10 @@ import axios from 'axios'
 
 import { batchActions } from '@webapp/store'
 
-import * as AppState from '@webapp/store/app/state'
-
 import * as autosave from '@webapp/app/components/autosave/actions'
 import ActionTypes from '@webapp/store/app/actions/actionTypes'
 import { ApiEndPoint } from '@common/api/endpoint'
+import { AppSelectors } from '@webapp/store/app/app.slice'
```
Comment (minotogna) [inline webapp/app/country/actions.ts:8]: we dont need AppSelectors, better hooks

File: webapp/app/countryLanding/views/messageBoard/messageBoardPanel.tsx [hunk 5]
```diff
@@ -1,15 +1,15 @@
 import React, { useEffect, useRef } from 'react'
-import { connect, useSelector } from 'react-redux'
+import { connect } from 'react-redux'
 import * as R from 'ramda'
 
 import Icon from '@webapp/components/icon'
-import useI18n from '@webapp/components/hooks/useI18n'
+import useI18n from '@webapp/store/app/hooks/useI18n'
```
Comment (minotogna) [inline webapp/app/countryLanding/views/messageBoard/messageBoardPanel.tsx:6]: use webapp/hooks

File: webapp/app/countryLanding/views/messageBoard/users.tsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { useCountryIso } from '@webapp/components/hooks'
-import useI18n from '@webapp/components/hooks/useI18n'
+import useI18n from '@webapp/store/app/hooks/useI18n'
```
Comment (minotogna) [inline webapp/app/countryLanding/views/messageBoard/users.tsx:3]: as above

File: webapp/store/app/app.slice.ts [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+      .addCase(initApp.rejected, (state, { payload }) => {
+        state.regionGroups = payload.regionGroups
+        state.loaded = true
+      })
+  },
+})
+
+export const { updateCountries, updateRegions, updateRegionGroups } = appSlice.actions
+
+export const AppSelectors = {
```
Comment (minotogna) [inline webapp/store/app/app.slice.ts:140]: we dont need this. better to use hooks to get data from store

File: webapp/store/app/hooks/index.ts [hunk 8]
```diff
@@ -1,21 +1,19 @@
...
-  return useSelector(AppState.getAssessmentType) as AssessmentType
-}
+export const useAssessmentType = (): AssessmentType => useAppSelector(AppSelectors.selectAssessmentType)
 
 export const useCountries = (): Array<Country> => {
   const i18n = useI18n()
-  const dispatch = useDispatch()
-  const countries = useSelector(AppState.getCountries) as Array<Country>
+  const dispatch = useAppDispatch()
+  const countries = useAppSelector(AppSelectors.selectCountries)
```
Comment (minotogna) [inline webapp/store/app/hooks/index.ts:16]: add hook.

File: webapp/store/app/app.slice.ts [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+import { applicationError } from '@webapp/components/error/actions'
+import * as UserState from '@webapp/store/user/state'
+
+import { createSlice, PayloadAction, createAsyncThunk } from '@reduxjs/toolkit'
+import { RootState } from '@webapp/store/RootState'
+import { AssessmentType } from '@core/assessment'
+import { i18n } from 'i18next'
+import { User } from '@core/auth'
+
+const initApp = createAsyncThunk(
```
Comment (minotogna) [inline webapp/store/app/app.slice.ts:16]: extract the actions in a separate file / separate files

**Comment by sorja** [comment]
PR closed; too much effort to find out reason why initCountry doesn't dispatch.
New pr based on this opening soon

**Pull request closed without merging**